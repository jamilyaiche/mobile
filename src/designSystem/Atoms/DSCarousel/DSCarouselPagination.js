import React from 'react';
import {
  View,
} from 'react-native';
import PropTypes from 'prop-types';
import Colors from "../../../assets/styles/Colors"
import Style from "../../../assets/styles/Style"

const DSCarouselPagination = (props) => {
  const {
    total,
    currentPage,
  } = props

  const renderDots = () => {
    const Dots = []
    for (let i = 0; i < total; i += 1) {
      const isActive = currentPage === i + 1

      const additionalStyle = {
        backgroundColor: isActive ? Colors.primary : null,
        marginLeft: i === 0 ? 0 : 13,
      };

      Dots.push((
        <View
          key={`pagination-dot-${i}`}
          style={[
            Style.carouselPagerDot,
            additionalStyle,
          ]}
        />
      ));
    }
    return Dots;
  }

  return (
    <View
      style={[Style.carouselPagerContainer,props.customConatainerStyle]}
    >
      {renderDots()}
    </View>
  )
}

DSCarouselPagination.propTypes = {
  currentPage: PropTypes.number.isRequired,
  total: PropTypes.number.isRequired,
  style: PropTypes.object,
  customConatainerStyle:PropTypes.object
}

DSCarouselPagination.defaultProps={
  customConatainerStyle:null
}

export default DSCarouselPagination;
