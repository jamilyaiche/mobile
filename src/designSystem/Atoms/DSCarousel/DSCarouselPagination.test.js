import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import DSCarouselPagination from "./DSCarouselPagination"
import {View} from "react-native"
jest.mock('rn-fetch-blob', () => {
    return {
      fs: {
        dirs: {
          DocumentDir: ''
        },
        writeFile: () => Promise.resolve()
      }
    }
  })
  jest.mock('react-native-x2-carousel', () => {
    return {
      fs: {
        dirs: {
          DocumentDir: ''
        },
        writeFile: () => Promise.resolve()
      }
    }
  })
describe("test carousel paging",()=>{
    it("carousel paging snapshot",()=>{
        const wrapper=renderer.create(<DSCarouselPagination total={3} currentPage={1} />)
        .toJSON()
        expect(wrapper).toMatchSnapshot()
    })

    it('test carousel paging current page',()=>{
        const wrapper=renderer.create(
            <DSCarouselPagination total={3} currentPage={1} />
        )
        expect(wrapper.root.findAllByType(View)[1].props.style[1].backgroundColor!==null).toEqual(true)
        expect(wrapper.root.findAllByType(View)[2].props.style[1].backgroundColor===null).toEqual(true)
    })
})