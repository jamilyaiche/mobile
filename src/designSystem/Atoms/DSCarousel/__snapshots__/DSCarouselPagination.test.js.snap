// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test carousel paging carousel paging snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "bottom": 40,
        "flexDirection": "row",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
      },
      null,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "borderColor": undefined,
          "borderRadius": 5,
          "borderWidth": 1.5,
          "height": 10,
          "width": 10,
        },
        Object {
          "backgroundColor": undefined,
          "marginLeft": 0,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderColor": undefined,
          "borderRadius": 5,
          "borderWidth": 1.5,
          "height": 10,
          "width": 10,
        },
        Object {
          "backgroundColor": null,
          "marginLeft": 13,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderColor": undefined,
          "borderRadius": 5,
          "borderWidth": 1.5,
          "height": 10,
          "width": 10,
        },
        Object {
          "backgroundColor": null,
          "marginLeft": 13,
        },
      ]
    }
  />
</View>
`;
